modules/plottinggl/processors/parallelcoordinates/parallelcoordinates.cpp

```diff
@@ -330,16 +179,55 @@ ParallelCoordinates::ParallelCoordinates()
     labelPosition_.addOption("below", "Below", LabelPosition::Below);
     labelPosition_.setSelectedIndex(2);
 
+    autoMargins_.onChange([&]() {
+        float left = 0;
+        float right = 0;
+        float top = 0;
+        float bottom = 0;
+
+        float leftLabelWidth = 0;
+        float rightLabelWidth = 0;
+        size_t maxLabelHeight = 0;
+
+        for (auto &p : axisVector_) {
+            if (p->isChecked()) {
+                if (labelPosition_.get() != LabelPosition::None) {
+                    if (leftLabelWidth == 0) {
+                        leftLabelWidth = p->labelTexture_->getWidth() / 2.f;
+                    }
+                    rightLabelWidth = p->labelTexture_->getWidth() / 2.f;
+                    maxLabelHeight = std::max(maxLabelHeight, p->labelTexture_->getHeight());
+                }
+                if (showValue_.get()) {
+                    right = static_cast<float>(p->minValTexture_->getWidth());
+                    right = static_cast<float>(p->maxValTexture_->getWidth());
+                }
+            }
+        }
+        if (labelPosition_.get() == LabelPosition::Above) {
+            top = static_cast<float>(maxLabelHeight);
+        } else if (labelPosition_.get() == LabelPosition::Below) {
+            bottom = static_cast<float>(maxLabelHeight);
+        }
+
+        left = std::max(leftLabelWidth, handleSize_.get().x / 2.f);
+        right = std::max(rightLabelWidth, right + handleSize_.get().x / 2.f);
+        top += handleSize_.get().y;
+        bottom += handleSize_.get().y;
+        margins_.setMargins(top + 1, right + 1, bottom + 1, left + 1);
+        // plus 1 to avoid text at the directly at the borders
+    });
+
     TransferFunction tf;
     tf.clear();
-    tf.add(0.0, vec4(1, 0, 0, 1));
-    tf.add(0.5, vec4(1, 1, 0, 1));
-    tf.add(1.0, vec4(0, 1, 0, 1));
+    tf.add(0.0f, vec4(1, 0, 0, 1));
```

Comment:
the first param of `TFPrimitiveSet` is a double.

Reply:
Strange that I didn't get a warning

Reply:
float to double should not cause any warning :)